include/pacemaker.h

```diff
@@ -55,16 +55,15 @@ int pcmk_fence_action(stonith_t *st, const char *target, const char *action,
  * \param[in]     st        A connection to the STONITH API.
  * \param[in]     target    The node to get history for.
  * \param[in]     timeout   How long to wait for the operation to complete (in ms).
- * \param[in]     quiet     Suppress most output.
  * \param[in]     verbose   Include additional output.
  * \param[in]     broadcast Gather fencing history from all nodes.
  * \param[in]     cleanup   Clean up fencing history after listing.
  *
  * \return Standard Pacemaker return code
  */
 int pcmk_fence_history(xmlNodePtr *xml, stonith_t *st, char *target,
-                       unsigned int timeout, bool quiet, int verbose,
-                       bool broadcast, bool cleanup);
+                       unsigned int timeout, int verbose, bool broadcast,
+                       bool cleanup);
```

Comment:
Yeah, me too.  It might be a good idea to go through all those other arguments and make them justify themselves.  Does verbose really need to be there, for instance, or can it also get pulled into the output object?

Reply:
It just occurred to me that this is the (future) public API (i.e. no output object), so quiet would need to be on the command line on this one, and it would set it after opening the output object internally.

Reply:
Oh yeah, I'll add it back in.  `quiet` can be set on the output object pretty much as needed, as `is_quiet` will check it every time.  You can basically set and unset it between each line of output if you really wanted to.  So it's no big deal to just set that in `pcmk_fence_history`.

Condensing verbose and quiet into a single parameter also sticks out to me.

Reply:
> Condensing verbose and quiet into a single parameter also sticks out to me.

Yeah that's always bugged me but unfortunately in existing usage they are logically separate. In some cases it's meaningful to specify both together to get particular output! On top of that, verbose is used for two distinct purposes -- to increase the level of detail the tool gives to the user, and to turn on debugging messages (mainly for developers).

We could potentially change it at a minor-version bump (e.g. 2.1.0). We could use a single verbosity integer, and make --verbose do +1 and --quiet do -1, with 1 (and maybe 2) being reserved for user verbosity and larger values for debugging, and/or we could create a separate --debug=N parameter that takes over the debugging control.